File: 763.划分字母区间.go
/*
 * @lc app=leetcode.cn id=763 lang=golang
 *
 * [763] 划分字母区间
 */
package main

import "fmt"

// @lc code=start
func partitionLabels(S string) []int {
	localtion := make(map[string][]int, len(S))

	for i := 0; i < len(S); i++ {
		localtion[string(S[i])] = append(localtion[string(S[i])], i)
	}

	for i := 0; i < len(map); i++ {}

	fmt.Println(localtion)
	return []int{3}
}

// @lc code=end

File: 763.划分字母区间.php
/*
 * @lc app=leetcode.cn id=763 lang=php
 *
 * [763] 划分字母区间
 */
<?php
// @lc code=start
class Solution {

    /**
     * @param String $S
     * @return Integer[]
     */
    function partitionLabels($S) {
        $location = [];
        for ($i = 0; $i < strlen($S); $i++) {
            $location[$S[$i]][] = $i;
        }

        $result = [0];
        $end = 0;
        foreach ($location as $v) {
            if ($end <= $v[0]) {
                $end = end($v);
                $result[] = $end - end($result) + 1;
            } else {
                if (end($v) > $end) $end = end($v);
            }
        }

        $cnt = count($result);
        if ($end > $result[$cnt-1] + $result[$cnt-2] - 1) {
            array_pop($result);
            $result[] = $end - end($result) + 1;
        }
        array_shift($result);

        return $result;
    }
}
// @lc code=end

